### PR TITLE
Respect configuration when preparing project files

### PR DESCRIPTION
**Pull request by nadyaA**
Implement processing configuration specific files on build, run and debug. There was some initial logic to process debug configuration by default, but when configuration is explicitly passed we did not respect it.  For example:
`tns build android --release`

Now if the passed argument is valid configuration we process specific files, taking the parameter into consideration.

https://github.com/NativeScript/nativescript-cli/issues/2934

common lib PR: https://github.com/telerik/mobile-cli-lib/pull/986

### CODE REVIEW

File: lib/commands/debug.ts [hunk 1]
```diff
@@ -66,8 +68,10 @@
 			skipWatcher: !this.$options.watch || this.$options.justlaunch,
 			watchAllFiles: this.$options.syncAllFiles
 		};
-
-		await this.$debugLiveSyncService.liveSync(deviceDescriptors, liveSyncInfo);
+const projectFilesConfig: IProjectFilesConfig = {
```
Comment (rosen-vladimirov) [inline lib/commands/debug.ts]: incorrect indentation

File: lib/commands/run.ts [hunk 2]
```diff
@@ -116,8 +116,10 @@ export class RunCommandBase implements ICommand {
 			watchAllFiles: this.$options.syncAllFiles,
 			clean: this.$options.clean
 		};
-
-		await this.$liveSyncService.liveSync(deviceDescriptors, liveSyncInfo);
+		const projectFilesConfig: IProjectFilesConfig = {
```
Comment (rosen-vladimirov) [inline lib/commands/run.ts]: these lines are introduced on a lot of places, maybe you can introduce a method in [helpers](https://github.com/telerik/mobile-cli-lib/blob/master/helpers.ts)

File: lib/commands/run.ts [hunk 3]
```diff
@@ -116,8 +116,10 @@ export class RunCommandBase implements ICommand {
...
+			configuration: this.$options.release ? Configurations.Release.toLowerCase() : Configurations.Debug.toLowerCase()
+		};
+		await this.$liveSyncService.liveSync(deviceDescriptors, liveSyncInfo, projectFilesConfig);
```
Comment (rosen-vladimirov) [inline lib/commands/run.ts]: Do you need to pass `projectFilesConfig` to `liveSync` method - `liveSync` is called only for debug configuration (you can check some lines above that in case `--release` is passed, we'll never get here. Isn't the default behavior of file mappings (using the debug configuration) working fine in this case?

File: lib/definitions/livesync.d.ts [hunk 4]
```diff
@@ -154,7 +154,7 @@ interface ILiveSyncService {
 	 * @param {ILiveSyncInfo} liveSyncData Describes the LiveSync operation - for which project directory is the operation and other settings.
 	 * @returns {Promise<void>}
 	 */
-	liveSync(deviceDescriptors: ILiveSyncDeviceInfo[], liveSyncData: ILiveSyncInfo): Promise<void>;
+	liveSync(deviceDescriptors: ILiveSyncDeviceInfo[], liveSyncData: ILiveSyncInfo, projectFilesConfig: IProjectFilesConfig): Promise<void>;
```
Comment (rosen-vladimirov) [inline lib/definitions/livesync.d.ts]: This method is exposed publicly when CLI is used as a library, is this really required?

File: lib/services/platform-service.ts [hunk 5]
```diff
@@ -325,7 +329,11 @@ export class PlatformService extends EventEmitter implements IPlatformService {
 			excludedDirs.push(constants.TNS_MODULES_FOLDER_NAME);
 		}
 
-		this.$projectFilesManager.processPlatformSpecificFiles(directoryPath, platform, excludedDirs);
+		const projectFilesConfig: IProjectFilesConfig = {
```
Comment (rosen-vladimirov) [inline lib/services/platform-service.ts]: this code is already available in this method - some lines above in the first `if`, can you declare it at a single place and use it in both methods where it is required.

**Comment by dtopuzov** [comment]
run ci

File: lib/commands/run.ts (hunk 3 above)
Comment (nadyaA) [inline lib/commands/run.ts]: 🎣

File: lib/tools/node-modules/node-modules-builder.ts [hunk 6]
```diff
@@ -43,6 +43,6 @@ export class NodeModulesBuilder implements INodeModulesBuilder {
 
 		return productionDependencies;
 	}
-}
+	}
```
Comment (rosen-vladimirov) [inline lib/tools/node-modules/node-modules-builder.ts]: Why is this changed?
